osu.Game/Beatmaps/Drawables/Cards/Buttons/FavouriteButton.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Framework.Logging;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Resources.Localisation.Web;
+
+namespace osu.Game.Beatmaps.Drawables.Cards.Buttons
+{
+    public class FavouriteButton : BeatmapCardIconButton, IHasCurrentValue<BeatmapSetFavouriteState>
+    {
+        private readonly BindableWithCurrent<BeatmapSetFavouriteState> current;
+
+        public Bindable<BeatmapSetFavouriteState> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        private readonly int onlineBeatmapID;
+
+        private PostBeatmapFavouriteRequest favouriteRequest;
+
+        public FavouriteButton(APIBeatmapSet beatmapSet)
+        {
+            current = new BindableWithCurrent<BeatmapSetFavouriteState>(new BeatmapSetFavouriteState(beatmapSet.HasFavourited, beatmapSet.FavouriteCount));
+            onlineBeatmapID = beatmapSet.OnlineID;
```

Comment:
not a huge issue, but any reason for only storing the onlineID here? generally we store the full model since there's no reason not to.

Reply:
Was in two minds about it, no strong opinion. I'll change back to storing full model anyways.